persistence-cassandra-4.0/src/main/java/io/stargate/db/cassandra/impl/Conversion.java

```diff
@@ -494,15 +492,12 @@ private static Result toResultInternal(
       case PREPARED:
         ResultMessage.Prepared prepared = (ResultMessage.Prepared) resultMessage;
         PreparedWithInfo preparedWithInfo = (PreparedWithInfo) prepared;
-        QueryHandler.Prepared preparedStatement =
-            QueryProcessor.instance.getPrepared(prepared.statementId);
         return new Result.Prepared(
             Conversion.toExternal(prepared.statementId),
             Conversion.toExternal(prepared.resultMetadataId),
             toResultMetadata(prepared.resultMetadata, null),
             toPreparedMetadata(
-                prepared.metadata.names,
-                preparedStatement.statement.getPartitionKeyBindVariableIndexes()),
+                prepared.metadata.names, preparedWithInfo.getPartitionKeyBindVariableIndexes()),
```

Comment:
I like this -- avoiding unnecessary separate preparation by bundling it with `PreparedWithInfo`.

Reply:
@ivansenic Should the similar change be done for 3.11 as well? Code as-is still has:

```
        ParsedStatement.Prepared preparedStatement =
            QueryProcessor.instance.getPrepared(prepared.statementId);
```

but may have access with `PreparedWithInfo`, or if not, could be added same as with 4.0.

Reply:
Yea very good catch, missed that somehow.